cmake_modules/IcebergThirdpartyToolchain.cmake

```
@@ -285,3 +287,20 @@ function(resolve_nlohmann_json_dependency)
 endfunction()
 
 resolve_nlohmann_json_dependency()
+
+# ----------------------------------------------------------------------
+# zlib
+
+function(resolve_zlib_dependency)
+  # use system zlib, zlib is required by arrow and avro
+  find_package(ZLIB REQUIRED)
+  if(ZLIB_FOUND)
+    list(APPEND ICEBERG_SYSTEM_DEPENDENCIES ZLIB)
+    message(STATUS "ZLIB_FOUND")
+    message(STATUS "ZLIB_LIBRARIES: ${ZLIB_LIBRARIES}")
+    message(STATUS "ZLIB_INCLUDE_DIR: ${ZLIB_INCLUDE_DIR}")
+  endif()
+
```

```suggestion
  set(ICEBERG_SYSTEM_DEPENDENCIES
      ${ICEBERG_SYSTEM_DEPENDENCIES}
      PARENT_SCOPE)
```

The CI is failed: https://github.com/apache/iceberg-cpp/actions/runs/15106986625/job/42457669021?pr=101 We need to add this to the parent scope